app-frontend/src/app/pages/projects/edit/annotate/annotate.controller.js

```diff
@@ -1,116 +1,506 @@
+        if (this.filterLabel.name !== 'All'
+            && this.filterLabel.name !== label
+        ) {
+            this.filterLabel.name = label;
+            this.onFilterChange(this.filterLabel);
+        } else if (this.filterLabel.name === label) {
+            this.onFilterChange(this.filterLabel);
+        }
+    }
+
+    onCancelUpdateAnnotation(annotation) {
+        this.disableToolbarAction = false;
+        this.isCreating = false;
+        this.disableSidebarAction = false;
+
+        this.editHandler.disable();
+
+        this.getMap().then((mapWrapper) => {
+            mapWrapper.setLayer(
+                'Annotation',
+                this.createLayerWithPopupFromGeojson(this.annoToExport),
+                true
+            );
+            mapWrapper.deleteLayers('draw');
+
+            this.updateFilterAndMapRender(annotation.properties.label);
+        });
+    }
+
+    /* eslint-disable no-underscore-dangle */
+    onUpdateAnnotationFinish(id, label, description) {
+        this.onAddAnnotation(id, label, description);
+    }
+    /* eslint-enable no-underscore-dangle */
+
+    /* eslint-disable no-underscore-dangle */
+    onDeleteAnnotation(id, label) {
+        _.remove(this.annoToExport.features, f => f.properties.id === id);
+
+        this.getMap().then((mapWrapper) => {
+            mapWrapper.setLayer(
+                'Annotation',
+                this.createLayerWithPopupFromGeojson(this.annoToExport),
+                true
+            );
+
+            this.labelInputs = this.updateLabelInputs(this.annoToExport.features);
+
+            if (!_.find(this.labelInputs, i => i.name === label)) {
+                this.filterLabel = {'name': 'All'};
+                this.onFilterChange(this.filterLabel);
+            }
+
+            this.updateFilterAndMapRender(label);
+
+            mapWrapper.deleteLayers('highlight');
+        });
+    }
+    /* eslint-enable no-underscore-dangle */
+
+    /* eslint-disable no-underscore-dangle */
+    onChangeAnnotationStroke(annotation) {
+        this.getMap().then((mapWrapper) => {
+            mapWrapper.deleteLayers('highlight');
+            mapWrapper.setLayer(
+                'highlight',
+                L.geoJSON(annotation, {
+                    style: () => {
+                        return {'color': '#ff5b4d', 'opacity': 0.8, 'fillOpacity': 0};
+                    },
+                    pointToLayer: (geoJsonPoint, latlng) => {
+                        return L.marker(latlng, {
+                            'icon': L.divIcon({'className': 'annotate-highlight-marker'})
+                        });
+                    }
+                }),
+                false
+            );
+            if (!_.isEmpty(mapWrapper.getLayers('highlight')[0]._layers)) {
+                mapWrapper.map.setView(
+                    mapWrapper.getLayers('highlight')[0].getBounds().getCenter()
+                );
+            }
+            this.$timeout(() => {
```

Comment:
if you click several annotations in the sidebar in a row, this timeout doesn't get reset. The timeout needs to get cancelled / not do anything if the annotation being selected has changed.
This will not be an issue if we change how clicking on the sidebar / map works